### PR TITLE
Start system-probe with the Agent also on upstart

### PR DESCRIPTION
**Pull request by albertvaka**
### What does this PR do?

Replicates the behaviour in https://github.com/DataDog/datadog-agent/pull/4883 for upstart, which was missing.

### Motivation

My previous PR implemented this logic on systemd but not on upstart.

### CODE REVIEW

**Review by leeavital — DISMISSED** [review]
I don't see any calls to setgid/setuid in these scripts, so I think these services all run as root.  If that's the case, I don't think we actually need this change since the system-probe will exit quickly if no system-probe.yaml exists already.

**Comment by albertvaka** [comment]
Even if the script runs as root, we can avoid running the system-probe if there is no reason for it. IMO the less things we run as root, the better.

**Review by KSerrania — DISMISSED** [review]

File: omnibus/config/templates/datadog-agent/upstart_debian.sysprobe.conf.erb [hunk 1]
```diff
@@ -22,7 +23,9 @@ pre-start script
 end script
 
 script
-  exec <%= install_dir %>/embedded/bin/system-probe --config=<%= etc_dir %>/system-probe.yaml --pid=<%= install_dir %>/run/system-probe.pid
+  if [ -f <%= etc_dir %>/system-probe.yaml ]; then
```
Comment (KSerrania) [inline omnibus/config/templates/datadog-agent/upstart_debian.sysprobe.conf.erb]: I think this condition should be run in the pre-start script:
```
pre-start script
  test -f /etc/system-probe.yaml || { stop ; exit 0; }

  ...
end script
```
It's more idiomatic (see [the Upstart doc](http://upstart.ubuntu.com/cookbook/#stop-a-job-from-running-if-a-pre-start-condition-fails)) and we don't try to mount `/sys/kernel/debug` for nothing (less things we run as root that way).

**Review by KSerrania — APPROVED** [review]
Tested on Ubuntu 14.04, `service datadog-agent-sysprobe status` reports `datadog-agent-sysprobe stop/waiting` unless the system-probe is enabled, and when the `system-probe.yaml` config file does not exist, the upstart logs say: `datadog-agent-sysprobe stop/pre-start, process 2566` as expected.

System-probe starts and stops with the Agent as expected.

**Pull request merged**